fiona/_shim2.pyx

```diff
@@ -27,23 +38,44 @@ cdef void* gdal_open_vector(char* path_c, int mode, drivers):
             if drv != NULL:
                 drvs = CSLAddString(drvs, name_c)
 
+    for k, v in options.items():
+        k = k.upper().encode('utf-8')
+        if isinstance(v, bool):
+            v = ('ON' if v else 'OFF').encode('utf-8')
+        else:
+            v = str(v).encode('utf-8')
+        log.debug("Set option %r: %r", k, v)
+        open_opts = CSLAddNameValue(open_opts, <const char *>k, <const char *>v)
+
```

Comment:
This section should be pulled out into a function and reused in `gdal_create`. I don't think there is a nice way to share it between shim2/shim22, but I still think it should come out.

Reply:
@snorfalorpagus I think I'm going to follow the rule of 3 here: it's okay to duplicate code 2x, but on the third time we refactor.